packages/back-end/src/util/mongo.util.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * Assists in migrating any field options that MongoDB has changed between major versions 3 and 4.
+ * Replaces the query with an equivalent where keys that can be mapped 1-to-1 are replaced with their new values.
+ *
+ * Docs referenced:
+ *  - v3 fields: https://mongodb.github.io/node-mongodb-native/3.7/reference/connecting/connection-settings/
+ *  - v4 fields: https://mongodb.github.io/node-mongodb-native/4.16/interfaces/MongoClientOptions.html
+ *
+ * @return ResultDeprecatedKeysMigrationV3to4
+ *
+ * Unsupported deprecated fields that do not have 1-to-1 mappings.
+ * Some fields no longer exist, while others are marked as deprecated in v4, which means they *could* be removed in the future.
+ *
+ * Note: Some v3 options were marked as deprecated but were included in v4 and not marked as deprecated. These have been ignored.
+ *
+ *  - autoReconnect: is no longer documented
+ *  - reconnectTries: is no longer documented
+ *  - reconnectInterval: is no longer documented
+ *  - ha: is no longer documented
+ *  - haInterval: is no longer documented
+ *  - secondaryAcceptableLatencyMS: is no longer documented. Possibly related: maxStalenessSeconds
+ *  - acceptableLatencyMS: is no longer documented. Possibly related: maxStalenessSeconds
+ *  - connectWithNoPrimary: is no longer documented
+ *  - w: still exists but is marked as deprecated -> writeConcern (incompatible types)
+ *  - j: is no longer documented (journal write concern). -> writeConcern (incompatible types)
+ *  - domainsEnabled: is no longer documented
+ *  - bufferMaxEntries: is no longer documented
+ *  - promiseLibrary: still exists but is marked as deprecated.
+ *  - loggerLevel: still exists but is marked as deprecated.
+ *  - logger: still exists but is marked as deprecated.
+ */
+export const getConnectionStringWithDeprecatedKeysMigratedForV3to4 = (
+  uri: string
+): ResultDeprecatedKeysMigrationV3to4 => {
+  const unsupportedV3FieldsInV4 = [
+    "autoReconnect",
+    "reconnectTries",
+    "reconnectInterval",
+    "ha",
+    "haInterval",
+    "secondaryAcceptableLatencyMS",
+    "acceptableLatencyMS",
+    "connectWithNoPrimary",
+    "j",
+    "domainsEnabled",
+    "bufferMaxEntries",
+  ];
+  const v3to4Mappings: Record<string, string> = {
+    poolSize: "maxPoolSize",
+    tlsinsecure: "tlsInsecure",
+
+    /**
+     * @deprecated
+     */
+    wtimeout: "wtimeoutMS",
+
+    appname: "appName",
+  };
+
+  try {
+    const remapped: string[] = [];
+    const parsedUrl = new URL(uri);
```

Comment:
Mongo connection strings can contain multiple comma-separated hosts. See replicaset example from their docs below.  It looks like this code will fall into the catch block for that, which is fine (we aren't trying to catch 100% of use cases, just the common ones). We should add a test case for this though.

https://www.mongodb.com/docs/manual/reference/connection-string/